### PR TITLE
add workflow_dispatch to build

### PR DESCRIPTION
**Pull request by daniell-canva**
Add workflow_dispatch to the build pipeline, which allows manual triggering.

Also add missing dependencies and fix a broken test to unblock the build pipeline.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,10 +14,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
 
-    - name: Set up Go 1.17
+    - name: Set up Go 1.19
```
Comment (jsing-canva) [inline .github/workflows/build.yml]: Hrmmm, 1.19 is no longer supported - are we able to get this to 1.21 or 1.22?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -13,10 +14,10 @@ jobs:
...
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17
+        go-version: 1.19
```
Comment (mallyx3) [inline .github/workflows/build.yml]: Could we use version 1.22? It's the current version in the infrastructure repo.

**Review by mallyx3 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (daniell-canva) [reply .github/workflows/build.yml]: Good catch, have updated to 1.22

File: .github/workflows/build.yml (hunk 2 above)
Comment (daniell-canva) [reply .github/workflows/build.yml]: Good catch, have updated to 1.22

**Pull request merged**